src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataQueueDiagnosticTagManager.java

```diff
@@ -0,0 +1,152 @@
+
+  /**
+   * Tag control object used for getting the value of the queue diagnostic tag for forcing a reset
+   * of the time tracker.
+   */
+  private static TagControl queueDiagnosticForceResetTag = null;
+
+  /**
+   * Tag control object used for updating the value of the queue diagnostic tag for the queue poll
+   * count.
+   */
+  private static TagControl queueDiagnosticPollCountTag = null;
+
+  /** Boolean flag indicating whether the queue diagnostic tags have been configured. */
+  private static boolean isConfigured = false;
+
+  /**
+   * Returns a boolean flag indicating whether the queue diagnostic tags have been configured.
+   *
+   * @return true if the queue diagnostic tags have been configured, false otherwise
+   */
+  public static synchronized boolean isConfigured() {
+    return isConfigured;
+  }
+
+  /**
+   * Configures the queue diagnostic tags if the specified <code>queueDiagnosticTagsEnabled</code>
+   * boolean is true.
+   *
+   * @throws Exception if a diagnostic tag is missing and cannot be created
+   */
+  public static synchronized void configureQueueDiagnosticTags() throws Exception {
+    // Configure queue running behind diagnostic tag
+    queueDiagnosticRunningBehindSecondsTag =
+        tryCreateDiagnosticTag(
+            HistoricalDataConstants.QUEUE_DIAGNOSTIC_TAG_RUNNING_BEHIND_SECONDS_NAME,
+            HistoricalDataConstants.QUEUE_DIAGNOSTIC_TAG_RUNNING_BEHIND_SECONDS_DESC,
+            HistoricalDataConstants.QUEUE_DIAGNOSTIC_TAG_IO_SERVER,
+            HistoricalDataConstants.QUEUE_DIAGNOSTIC_TAG_RUNNING_BEHIND_SECONDS_TYPE);
+
+    // Configure queue reset tag
+    queueDiagnosticForceResetTag =
+        tryCreateDiagnosticTag(
+            HistoricalDataConstants.QUEUE_DIAGNOSTIC_TAG_FORCE_RESET_NAME,
+            HistoricalDataConstants.QUEUE_DIAGNOSTIC_TAG_FORCE_RESET_DESC,
+            HistoricalDataConstants.QUEUE_DIAGNOSTIC_TAG_IO_SERVER,
+            HistoricalDataConstants.QUEUE_DIAGNOSTIC_TAG_FORCE_RESET_TYPE);
+
+    // Configure queue poll count tag
+    queueDiagnosticPollCountTag =
+        tryCreateDiagnosticTag(
+            HistoricalDataConstants.QUEUE_DIAGNOSTIC_TAG_POLL_COUNT_NAME,
+            HistoricalDataConstants.QUEUE_DIAGNOSTIC_TAG_POLL_COUNT_DESC,
+            HistoricalDataConstants.QUEUE_DIAGNOSTIC_TAG_IO_SERVER,
+            HistoricalDataConstants.QUEUE_DIAGNOSTIC_TAG_POLL_COUNT_TYPE);
+
+    // Set configured flag
+    isConfigured = true;
+  }
+
+  /**
+   * Attempts to create a diagnostic tag with the specified tag name, description, IO server, and
+   * type.
+   *
+   * @param tagName diagnostic tag name
+   * @param tagDesc diagnostic tag description
+   * @param tagIoServer diagnostic tag IO server
+   * @param tagType diagnostic tag type
+   * @return corresponding {@link TagControl} object for the diagnostic tag
+   * @throws Exception if the diagnostic tag is missing and cannot be created
```

Comment:
It can throw and EWException if its created but cannot be accessed via TagControl()

Reply:
Added